org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -1863,7 +1863,7 @@ private void testCompletion_classMethodOverride(String projectName, boolean supp
 			expectedText.append("${0:");
 		}
 		expectedText.append("// TODO Auto-generated method stub\n\t");
-		expectedText.append("return super.toString();");
+		expectedText.append("throw new UnsupportedOperationException(\"Unimplemented method \'toString\'\");");
```

Comment:
For cases like this, would it be possible to retain this particular behaviour ? I think when overriding, it can be pretty reasonable to delegate the base class.

Reply:
I think if we wanted to add another template that contains that original behavior and use that for this case, we would have to modify this https://git.eclipse.org/r/plugins/gitiles/jdt/eclipse.jdt.ui/+/2ccac2274e025ae25d67c0016439c0b7fe669f3f/org.eclipse.jdt.ui/core%20extension/org/eclipse/jdt/internal/corext/codemanipulation/StubUtility.java#109 since it's where it decides which template is used. I'm not sure if there is an easier way to do it?

Reply:
Yeah, that's exactly where it looks like, unless there's some clever trick we can do to treat the constructor a bit differently. I think your reference is old though because the class has been moved into jdt.core.manipulation : https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/c9c7e9d3242e1211ac2fc25b91fb22f65203314a/org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/StubUtility.java#L162 .